File: Arrays/Reverse/reverseArray.cpp
#include<iostream>
using namespace std;

void reverseArray(int arr[], int n){

    int start = 0; 
    int end = n-1;
    while(start<end){
        int temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp; 
        start++;
        end--; 
    }
}

int main()
{
    int arr[] = {10,7,5,2,4,4,6,6,8,9};
    int n = sizeof(arr)/sizeof(arr[0]);
    reverseArray(arr,n);
    reverseArray(arr,n);
    for(int x: arr){
        cout<<x<<endl;
    }
    return 0;
}

File: Arrays/Reverse/reverseArray.java
package Arrays.Reverse;

public class reverseArray {

    public static Integer[] reverse(Integer[] arr){
        Integer start = 0; 
        Integer end = arr.length - 1; 
        while(start<end){
            Integer temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp; 
            start++;
            end--;
        }   
        return arr;
    }

    public static void main(String args[]){
        Integer arr[] = {10,7,5,2,4,4,6,6,8,9};
        arr = reverse(arr);
        for(Integer x: arr){
            System.out.println(x +" ");
        }
    } 
    
}
